chain-signatures/contract-sol/programs/chain-signatures-project/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::{self, Token};
+
+declare_id!("4kZoBXmUBLveRS3sboGF557tYsR7SzLDsWmP4sz7VQEs");
+
+#[program]
+pub mod chain_signatures_project {
+    use super::*;
+
+    pub fn initialize(
+        ctx: Context<Initialize>,
+        signature_deposit: u64,
+    ) -> Result<()> {
+        let program_state = &mut ctx.accounts.program_state;
+        program_state.admin = ctx.accounts.admin.key();
+        program_state.signature_deposit = signature_deposit;
+
+        Ok(())
+    }
+
+    pub fn sign(
+        ctx: Context<Sign>,
+        payload: [u8; 32],
+        key_version: u32,
+        path: String,
+        algo: String,
+        dest: String,
+        params: String,
+    ) -> Result<()> {
+        let program_state = &ctx.accounts.program_state;
+        let requester = &ctx.accounts.requester;
+        let system_program = &ctx.accounts.system_program;
+        
+        require!(
+            ctx.accounts.requester.lamports() >= program_state.signature_deposit,
+            ChainSignaturesError::InsufficientDeposit
+        );
+        
+        let transfer_instruction = anchor_lang::system_program::Transfer {
+            from: requester.to_account_info(),
+            to: program_state.to_account_info(),
+        };
+        
+        anchor_lang::system_program::transfer(
+            CpiContext::new(
+                system_program.to_account_info(),
+                transfer_instruction,
+            ),
+            program_state.signature_deposit,
+        )?;
+        
+        emit!(SignatureRequestedEvent {
+            sender: *requester.key,
+            payload,
+            key_version,
+            deposit: program_state.signature_deposit,
+            chain_id: 0,
```

Comment:
Will we use this id in the future for routing between chains? Should we maybe pick a number for Solana that's unlikely to collide with any potential future EVM chains, while also being different from uninitialized (0) ?

Reply:
it is taken from https://github.com/sig-net/chain-signatures-solana-stub/blob/6ae5c760278756ee84a407d7e19bc12fb9d8e963/chain-signatures-project/programs/chain-signatures-project/src/lib.rs#L98, @esaminu do you think we can use 0x800001f5 here (as you chosen for sig-server.ts)